src/components/infiniteScroll/index.js

```diff
@@ -1,35 +1,224 @@
+  hasMore: boolean,
+  initialLoad: boolean,
+  loader: React.Node,
+  loadMore: Function,
+  isLoadingMore: boolean,
+  pageStart: number,
+  threshold: number,
+  useWindow: boolean,
+  isReverse: boolean,
+  scrollElement: ?Object,
+  children: Array<?React.Node> | ?React.Node,
+  className: string,
+};
+
+export default class InfiniteScroll extends React.Component<Props> {
+  static defaultProps = {
+    element: 'div',
+    hasMore: false,
+    initialLoad: true,
+    pageStart: 0,
+    threshold: 250,
+    useWindow: true,
+    isReverse: false,
+    scrollElement: null,
+    scrollComponent: null,
+  };
+
+  scrollListener: Function;
+  scrollComponent: Object;
+  pageLoaded: number;
+  _defaultLoader: React.Node;
+
+  constructor(props: Props) {
+    super(props);
+
+    this._defaultLoader = props.loader;
+    this.scrollListener = this.scrollListener.bind(this);
+  }
+
+  componentDidMount() {
+    this.pageLoaded = this.props.pageStart;
+    this.attachScrollListener();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const curr = this.props;
+
+    if (curr.isLoadingMore) {
+      return;
     }
 
-    componentWillUnmount() {
-      // if (!this.scrollBody) return;
-      // let node = this.scrollBody;
-      // if (node) {
-      // 	node.removeEventListener('scroll', this.handleScroll)
-      // }
+    if (
+      Array.isArray(prevProps.children) &&
+      Array.isArray(curr.children) &&
+      prevProps.children.length === curr.children.length
+    ) {
+      return;
     }
 
-    handleScroll = () => {};
+    if (
+      React.Children.toArray(curr.children).length === 1 &&
+      React.Children.toArray(curr.children)[0].type.name ===
+        'FlipMovePropConverter'
+    ) {
+      const currFlipMoveChildren = React.Children.toArray(curr.children)[0]
+        .props.children;
+      const prevFlipMoveChildren = React.Children.toArray(prevProps.children)[0]
+        .props.children;
 
-    fetchMore = () => {
-      this.props.data.fetchMore();
-    };
+      if (
+        Array.isArray(prevFlipMoveChildren) &&
+        Array.isArray(currFlipMoveChildren) &&
+        prevFlipMoveChildren.length === currFlipMoveChildren.length
+      ) {
+        return;
+      }
+    }
+
+    this.attachScrollListener();
+  }
+
+  render() {
+    const {
+      children,
+      element,
+      hasMore,
+      initialLoad,
+      loader,
+      loadMore,
+      pageStart,
+      threshold,
+      useWindow,
+      isReverse,
+      scrollElement,
+      isLoadingMore,
+      ...props
+    } = this.props;
 
-    render() {
-      return (
-        <Comp
-          ref={scrollBody => (this.scrollBody = scrollBody)}
-          {...this.props}
-          fetchMore={this.fetchMore}
-        />
-      );
+    if (scrollElement) {
+      // $FlowFixMe
+      props.ref = node => {
+        this.scrollComponent = scrollElement;
+      };
+    } else {
+      // $FlowFixMe
+      props.ref = node => {
+        this.scrollComponent = node;
+      };
```

Comment:
And what are we doing here? Why are we attaching something to the props? This should be a callback and it should definitely not live in the render method:

```JS
this.props.innerRef((node) => {});
```

Reply:
This is an artifact from the original library we are using; happy to take it out, but I'm not sure why the original author had it there and what might break without it.

Reply:
I have no clue how to approach this to be honest man :/